store/redis_supplier.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package store
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"fmt"
+
+	"time"
+
+	l4g "github.com/alecthomas/log4go"
+	"github.com/go-redis/redis"
+	"github.com/mattermost/platform/model"
+)
+
+type RedisSupplier struct {
+	next   LayeredStoreSupplier
+	client *redis.Client
+}
+
+func GetBytes(key interface{}) ([]byte, error) {
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
+	err := enc.Encode(key)
+	if err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+func DecodeBytes(input []byte, thing interface{}) error {
+	dec := gob.NewDecoder(bytes.NewReader(input))
+	err := dec.Decode(thing)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func NewRedisSupplier() *RedisSupplier {
+	supplier := &RedisSupplier{}
+
+	supplier.client = redis.NewClient(&redis.Options{
+		Addr:     "localhost:6379",
```

Comment:
Yes, just for testing and dev. I don't want customer's trying it before it's ready.